File: tests/benchmarks/resolution/fixtures/kotlin/Service.kt
package benchmark

class UserService(private val repo: UserRepository) {

    fun createUser(name: String, email: String): Boolean {
        if (!validateName(name)) return false
        if (!validateEmail(email)) return false
        val user = User(name, email)
        return repo.saveIfValid(user)
    }

    fun getUser(name: String): User? {
        return repo.findByName(name)
    }

    fun removeUser(name: String): Boolean {
        return repo.delete(name)
    }
}

fun buildService(): UserService {
    val repo = UserRepository()
    return UserService(repo)
}

File: tests/benchmarks/resolution/fixtures/python/service.py
from models import User
from repository import create_repository


def validate_email(email):
    return "@" in email and "." in email


class UserService:
    def __init__(self, repo):
        self.repo = repo

    def create_user(self, user_id, name, email):
        if not validate_email(email):
            raise ValueError("Invalid email")
        user = User(user_id, name, email)
        if not user.validate():
            raise ValueError("Invalid user data")
        self.repo.save(user)
        return user

    def get_user(self, user_id):
        return self.repo.find_by_id(user_id)

    def remove_user(self, user_id):
        return self.repo.delete(user_id)


def build_service():
    repo = create_repository()
    return UserService(repo)
